pkg/orchestrator/endpoint.go

```diff
@@ -47,6 +47,8 @@ func NewBaseEndpoint(params *BaseEndpointParams) *BaseEndpoint {
 func (e *BaseEndpoint) SubmitJob(ctx context.Context, request *SubmitJobRequest) (_ *SubmitJobResponse, err error) {
```

Comment:
The updatejob method used here should not change the paramter passed to it, source of future bugs.